pipelineWrapper.py

```diff
@@ -0,0 +1,157 @@
+    def _prepare_mount(self):
+        # Get name of most recent running container. If socket is mounted, should be this one.
+        name_command = ['docker', 'ps', '--format', '{{.Names}}']
+        try:
+            name = subprocess.check_output(name_command).split('\n')[0]
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError('No container detected, ensure Docker is being run with: '
+                               '"-v /var/run/docker.sock:/var/run/docker.sock" as an argument.'
+                               ' \n\n{}'.format(e.message))
+        # Get name of mounted volume
+        blob = json.loads(subprocess.check_output(['docker', 'inspect', name]))
+        mounts = blob[0]['Mounts']
+        # Ensure docker.sock is mounted correctly
+        sock_mnt = [x['Source'] == x['Destination'] for x in mounts if 'docker.sock' in x['Source']]
+        require(len(sock_mnt) == 1, 'Missing socket mount. Requires the following: '
+                                      'docker run -v /var/run/docker.sock:/var/run/docker.sock')
+        # Ensure formatting of command for 2 mount points
+        if len(mounts) == 2:
+            require(all(x['Source'] == x['Destination'] for x in mounts),
+                    'Docker Src/Dst mount points, invoked with the -v argument, '
+                    'must be the same if only using one mount point aside from the docker socket.')
+            work_mount = [x['Source'] for x in mounts if 'docker.sock' not in x['Source']]
+        else:
+            # Ensure only one mirror mount exists aside from docker.sock
+            mirror_mounts = [x['Source'] for x in mounts if x['Source'] == x['Destination']]
+            work_mount = [x for x in mirror_mounts if 'docker.sock' not in x]
+            require(len(work_mount) == 1, 'Wrong number of mirror mounts provided, see '
+                                          'documentation.')
+        # Output log information
+        log.info('The work mount is: %s', work_mount[0])
+        self._mount = work_mount[0]
+        return self._mount
+
+    def _make_command_prefix(self, jobstore_path, config_path, workdir_path):
+        return [self._name, 'run', jobstore_path,
+                '--config', config_path,
+                '--workDir', workdir_path,
+                '--retryCount', '1']
+
+    def _create_workdir(self, args):
+        if os.path.exists(self._workdir):
+            if self._resume and args.resume:
+                 log.info('Reusing temporary directory: %s', self._workdir)
+            else:
+                raise UserError('Temporary directory {} already exists. Run with --resume option or'
+                                ' remove directory.'.format(self._workdir))
+        else:
+            os.makedirs(self._workdir)
+            log.info('Temporary directory created: %s', self._workdir)
+
+class UserError(Exception):
+    pass
+
+
+class DefinitionError(Exception):
+    pass
+
+
+def require(expression, message):
```

Comment:
`require` is already in Toil-lib, just import it:   `from toil_lib import require`